executor/tests/test_executor.py

```diff
@@ -51,8 +51,8 @@ def test_index(tmpdir):
         uses_metas=metas,
     )
     with f:
-        result = f.post(on='/index', inputs=docs, return_results=True)
-        assert sum([len(r.docs) for r in result]) == N
+        docs = f.post(on='/index', inputs=docs, return_results=True)
+        assert len(docs) == N
```

Comment:
move this check to outside the `context manager`.  Otherwise u could have a false positive

Reply:
In which case,  a false positive occurs?

Reply:
If the context manager leaves before getting in the assert may have problems

Reply:
I see